From 6a86bde1c22dd8c71056968405d0d8b56723aa9e Mon Sep 17 00:00:00 2001
From: Zim <zezeozue@google.com>
Date: Fri, 4 Dec 2020 11:20:02 +0000
Subject: [PATCH] Block SAF directory access to /sdcard/Android

This works for target R+ apps, but need to come up with a better story
for target <R. The enforcement in DocumentsUI is target SDK based
and those apps bypass the block_tree flag.

Bug: 157320644
Test: atest DocumentsTest
Change-Id: I93aea7cd5d3b7befea1f78e1cf3f31f8250f1d33
(cherry picked from commit 9d508579aed406220c11022cfc983da46396bd07)
Merged-In: I93aea7cd5d3b7befea1f78e1cf3f31f8250f1d33
(cherry picked from commit 1ea98d44490a2383a604b546a5671a783c1000dd)
---
 .../com/android/externalstorage/ExternalStorageProvider.java | 5 +++++
 1 file changed, 5 insertions(+)

diff --git a/packages/ExternalStorageProvider/src/com/android/externalstorage/ExternalStorageProvider.java b/packages/ExternalStorageProvider/src/com/android/externalstorage/ExternalStorageProvider.java
index 11d1b0a9ef2a..087275e73ee8 100644
--- a/packages/ExternalStorageProvider/src/com/android/externalstorage/ExternalStorageProvider.java
+++ b/packages/ExternalStorageProvider/src/com/android/externalstorage/ExternalStorageProvider.java
@@ -322,6 +322,11 @@ public class ExternalStorageProvider extends FileSystemProvider {
                 return true;
             }

+            if (TextUtils.equals(Environment.DIRECTORY_ANDROID.toLowerCase(),
+                    path.toLowerCase())) {
+                return true;
+            }
+
             return false;
         } catch (IOException e) {
             throw new IllegalArgumentException(
--
2.17.1
